src/Dependencies/Collections/SegmentedArray.cs

```diff
@@ -38,6 +41,315 @@ internal static void Copy<T>(SegmentedArray<T> sourceArray, SegmentedArray<T> de
             }
         }
 
+        public static void Copy<T>(SegmentedArray<T> sourceArray, Array destinationArray, int length)
+        {
+            if (destinationArray is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.destinationArray);
+            if (length == 0)
+                return;
+
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length));
+            if (length > sourceArray.Length)
+                throw new ArgumentException(SR.Arg_LongerThanSrcArray, nameof(sourceArray));
+            if (length > destinationArray.Length)
+                throw new ArgumentException(SR.Arg_LongerThanDestArray, nameof(destinationArray));
+
+            var copied = 0;
+            foreach (var memory in sourceArray.GetSegments(0, length))
+            {
+                if (!MemoryMarshal.TryGetArray<T>(memory, out var segment))
+                {
+                    throw new NotSupportedException();
+                }
+
+                Array.Copy(segment.Array!, segment.Offset, destinationArray, copied, segment.Count);
```

Comment:
>copied, segment.Count [](start = 77, length = 21)

Consider using named parameters here, since this is multiple int parameters in a row.

Reply:
Similar comments below.

---
In reply to: [574106770](https://github.com/dotnet/roslyn/pull/51064#discussion_r574106770) [](ancestors = 574106770)

Reply:
➡️ Now updated